### PR TITLE
Displaying live data from the api

### PR DESCRIPTION
**Pull request by BenCampbellNelson**
I have set up the widget component with a function that fetches each endpoint. As each endpoint is currently displaying the most recent data I decided to have live most recent data displaying be the main priority.

 Next steps are going to be fetching an array of all recent data and begin displaying that across a daily/weekly/monthly graph. I removed the expand widget button and decided to make the whole widget clickable for expansion as will make more sense for a user to just tap/click on a widget instead of trying to find a little button each time. That functionality isn't present on the frontend as there is a check for graph components (which there aren't currently).

### CODE REVIEW

**Comment by SamanthaCra** [comment]
@BenCampbellNelson I have pulled the branch down with dummy data in the database, I am only getting the following:
image

File: fullstack/src/app/page.js [hunk 1]
```diff
@@ -5,38 +5,19 @@ import LineChartComponent from "@/components/graphs/LineChartComponent";
...
-        <Widget name={'Air Pressure'} data={airPressureData} GraphComponent={BarChartComponent} datakey="pressure"/>
-        <Widget name={'Humidity'} data={'coming soon'} GraphComponent={null} />
-        <Widget name={'Wind'} data={'coming soon'} GraphComponent={null} />
+        <Widget name="Temperature" dataKey="temperature" GraphComponent={null} />
+        <Widget name="Air Pressure" dataKey="pressure" GraphComponent={null} />
+        <Widget name="Humidity" dataKey="humidity" GraphComponent={null} />
+        <Widget name="Wind" dataKey="wind" GraphComponent={null} />
+        <Widget name="CO2" dataKey="co2" GraphComponent={null} />
+        <Widget name="Gas" dataKey="gas" GraphComponent={null} />
+        <Widget name="Dust" dataKey="dust" GraphComponent={null} />
```
Comment (SamanthaCra) [inline fullstack/src/app/page.js:18]: This is missing rain but I understand it is also not in the routes. We can add this next merge.

File: fullstack/src/components/widget.jsx [hunk 2]
```diff
@@ -1,70 +1,99 @@
...
   const toggleExpand = () => {
-    if (latestData) { // Only allow toggle if data exists
+    if (data) {
       setIsExpanded(!isExpanded);
     }
   };
 
-  const latestData =
-    Array.isArray(data) && data.length > 0 ? data[data.length - 1] : null;
+  const latestData = data?.length ? data[data.length - 1] : null; // Get the latest data point
```
Comment (SamanthaCra) [inline fullstack/src/components/widget.jsx:42]: The routes grab the latest one data point at the moment, is this needed?

**Review by SamanthaCra — APPROVED** [review]
Nice work, thanks for fixing the issues that came up when running it. Just a few questions but otherwise okay.

File: fullstack/src/components/widget.jsx (hunk 2 above)
Comment (BenCampbellNelson) [reply fullstack/src/components/widget.jsx:42]: I thought for future we may have the routes fetching an array of more data for displaying on graphs, daily view etc. SO having this means not needing to make an adjustment down the track

**Review by fuzzyers — CHANGES_REQUESTED** [review]
image
Data checking for null data needs to be more robust in the widgets to avoid issues such as these.

Is container.js component no longer being used?

**Review by fuzzyers — APPROVED** [review]
Awesome problem is fixed you can merge this now

**Pull request merged**